src/base/utils/fs.cpp

```diff
@@ -518,3 +518,10 @@ QString Utils::Fs::cacheLocation()
         locationDir.mkpath(locationDir.absolutePath());
     return location;
 }
+
+QString Utils::Fs::tempPath()
+{
+    static const QString path = QDir::tempPath() + "/.qBittorrent/";
+    QDir().mkdir(path);
```

Comment:
May I suggest something like the following to keep temporary dir management  local?
```C++
struct MyTempDir {
     MyTempDir() {
          : path(QDir::tempPath() + "/.qBittorrent/") 
      {
        QDir().mkdir(path);
        assert(QDir().exists(path)); // or whatever check is appropriate (log, exception...)
      }
      ~MyTempDir()
     {
        Utils::Fs::removeDirRecursive(path);
     }
     QString path;
};
static const MyTempDir tempDir;
return tempDir.path;
```